### PR TITLE
ENH: Overwrite attr's string conversion dunders

### PR DESCRIPTION
**Pull request by oesteban**
Helps I/O of the config file being drafted for fMRIPrep (https://github.com/poldracklab/fmriprep/pull/2018)

### CODE REVIEW

File: niworkflows/utils/spaces.py [hunk 1]
```diff
@@ -219,6 +219,19 @@ def _check_name(self, attribute, value):
...
+        Format this reference.
+
+        Examples
+        --------
+        >>> str(Reference(space='MNIPediatricAsym', spec={'cohort': 2, 'res': 1}))
+        'MNIPediatricAsym:cohort-2:res-1'
+
+        """
+        return ':'.join([self.space] + [
+            '-'.join((k, str(v))) for k, v in sorted(self.spec.items())])
```
Comment (mgxd) [inline niworkflows/utils/spaces.py]: Nothing is coming to mind, but could there be some unintended consequence of sorting these?

Comment (oesteban) [reply niworkflows/utils/spaces.py]: the intended consequence is that `cohort` goes always first. This is going to be problematic if we add new specs that would sort before `cohort` - but I don't see that happening anytime soon.

**Comment by oesteban** [comment]
> This still doesn't solve the I/O problem when there are 2 specifications of the same template, i.e.
> 
> `MNI152NLin2009cAsym:res-2:res-native`

Yes, it does. A `Space` object just can't/shouldn't have two resolution specs. The `SpatialReference.fromstring` method will make sure the dot product of specs is done, and one `Space` for each is generated.

File: niworkflows/utils/spaces.py (hunk 1 above)
Comment (mgxd) [reply niworkflows/utils/spaces.py]: @oesteban when I try running poldracklab/fmriprep#2018 with this branch and

`--output-spaces MNI152NLin2009cAsym:res-2:res-native fsaverage5 fsaverage:den-30k`

I'm getting the following traceback:
```
Traceback (most recent call last):
  File "/usr/local/miniconda/bin/fmriprep", line 10, in <module>
    sys.exit(main())
  File "/usr/local/miniconda/lib/python3.7/site-packages/fmriprep/cli/run.py", line 42, in main
    config.load(config_file)
  File "/usr/local/miniconda/lib/python3.7/site-packages/fmriprep/config.py", line 383, in load
    init_spaces()
  File "/usr/local/miniconda/lib/python3.7/site-packages/fmriprep/config.py", line 447, in init_spaces
    spaces = _SRs(
  File "/usr/local/miniconda/lib/python3.7/site-packages/niworkflows/utils/spaces.py", line 455, in __init__
    self.__iadd__(spaces)
  File "/usr/local/miniconda/lib/python3.7/site-packages/niworkflows/utils/spaces.py", line 466, in __iadd__
    self.append(space)
  File "/usr/local/miniconda/lib/python3.7/site-packages/niworkflows/utils/spaces.py", line 530, in append
    raise ValueError('space "%s" already in spaces.' % str(value))
ValueError: space "MNI152NLin2009cAsym" already in spaces.
```

Comment (oesteban) [reply niworkflows/utils/spaces.py]:
```
$ fmriprep /oak/stanford/groups/russpold/data/openfmri/ds000005 out/ participant --participant-label 01 -vv --nthreads 8 --omp-nthreads 8 -w fprep-work2 --skip-bids-validation --notrack --output-spaces MNI152NLin2009cAsym:res-2:res-native fsaverage5 fsaverage:den-30k
200305-11:13:51,377 nipype.workflow WARNING:
         Previous output generated by version 20.0.0rc2+1.g68ab3205 found.
200305-11:13:51,715 nipype.workflow IMPORTANT:
         
    Running fMRIPREP version 20.0.1+14.gfd8d3ea6:
      * BIDS dataset path: /oak/stanford/groups/russpold/data/openfmri/ds000005.
      * Participant list: ['01'].
      * Run identifier: 20200305-111338_3db7c50f-a41e-41e6-bf87-76336f3595e0.
      * Output spaces: MNI152NLin2009cAsym:res-2 MNI152NLin2009cAsym:res-native fsaverage:den-10k fsaverage:den-30k.
    
200305-11:13:52,306 nipype.utils INFO:
         Running nipype version 1.5.0-dev+g135d900b5 (latest: 1.4.2)
Process Process-2:
Traceback (most recent call last):
  File "/home/oesteban/.anaconda3/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
    self.run()
  File "/home/oesteban/.anaconda3/lib/python3.7/multiprocessing/process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "/home/oesteban/workspace/fmriprep/fmriprep/cli/workflow.py", line 74, in build_workflow
    retval['workflow'] = init_fmriprep_wf()
  File "/home/oesteban/workspace/fmriprep/fmriprep/workflows/base.py", line 75, in init_fmriprep_wf
    single_subject_wf = init_single_subject_wf(subject_id)
  File "/home/oesteban/workspace/fmriprep/fmriprep/workflows/base.py", line 234, in init_single_subject_wf
    config.workflow.skull_strip_template)[0],
TypeError: init_anat_preproc_wf() got an unexpected keyword argument 'num_t1w'
```

And 
```
$ less fprep-work2/.fmriprep.toml 
[nipype]
crashfile_format = "txt"
get_linked_libs = false
nprocs = 8
omp_nthreads = 8
plugin = "MultiProc"
resource_monitor = false
stop_on_first_crash = false
version = "1.5.0-dev+g135d900b5"

[execution]
bids_dir = "/oak/stanford/groups/russpold/data/openfmri/ds000005"
boilerplate_only = false
exec_env = "posix"
fs_license_file = "/opt/freesurfer/license.txt"
fs_subjects_dir = "/home/oesteban/tmp/fmriprep-ds005/out/freesurfer"
layout = "BIDS Layout: ...usspold/data/openfmri/ds000005 | Subjects: 16 | Sessions: 0 | Runs: 48"
log_dir = "/home/oesteban/tmp/fmriprep-ds005/out/fmriprep/logs"
log_level = 15
low_mem = false
md_only_boilerplate = false
notrack = true
output_dir = "/home/oesteban/tmp/fmriprep-ds005/out"
output_spaces = "MNI152NLin2009cAsym:res-2 MNI152NLin2009cAsym:res-native fsaverage:den-10k fsaverage:den-30k"
reports_only = false
run_uuid = "20200305-111338_3db7c50f-a41e-41e6-bf87-76336f3595e0"
participant_label = [ "01",]
version = "20.0.1+14.gfd8d3ea6"
work_dir = "/home/oesteban/tmp/fmriprep-ds005/fprep-work2"
write_graph = false

[workflow]
anat_only = false
aroma_err_on_warn = false
aroma_melodic_dim = -200
bold2t1w_dof = 6
cifti_output = false
fmap_bspline = false
force_syn = false
hires = true
ignore = []
longitudinal = false
medial_surface_nan = false
run_reconall = true
skull_strip_fixed_seed = false
skull_strip_template = "OASIS30ANTs"
t2s_coreg = false
use_aroma = false

[nipype.plugin_args]
maxtasksperchild = 1
raise_insufficient = false
...
```

Comment (oesteban) [reply niworkflows/utils/spaces.py]: You might need to pull changes from the fmriprep branch?

**Pull request merged**